testinfra/host.py

```diff
@@ -29,14 +29,21 @@ def __repr__(self):
 
     def exists(self, command):
         """Return True if given command exist in $PATH"""
-        return self.run_expect([0, 1, 127], "command -v %s", command).rc == 0
+        return (
+            self.run_expect(
+                [0, 1, 127], "command -v %s || which %s", command, command
+            ).rc
+            == 0
+        )
```

Comment:
I think we sould run `which` only if `command` exit with 127 (not 1).

Also the implementation command (command or which) could be in a cached_property on the host object.